test/jdk/java/net/httpclient/SendResponseHeadersTest.java

```diff
@@ -40,46 +42,56 @@
 import java.net.InetAddress;
 import java.net.InetSocketAddress;
 import java.net.URI;
+import java.net.URISyntaxException;
 import java.net.http.HttpClient;
 import java.net.http.HttpRequest;
 import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.nio.charset.StandardCharsets;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 
 import static org.testng.Assert.expectThrows;
 import static org.testng.Assert.fail;
 
-
 public class SendResponseHeadersTest {
+    URI uri;
+    HttpServer server;
+    ExecutorService executor;
 
-    @Test
-    public void testSend() throws Exception {
-        var addr = new InetSocketAddress(InetAddress.getLoopbackAddress(), 0);
-        var server = HttpServer.create(addr, 0);
+    @BeforeTest
+    public void setUp() throws IOException, URISyntaxException {
+        var loopback = InetAddress.getLoopbackAddress();
+        var addr = new InetSocketAddress(loopback, 0);
         var path = "/test/foo.html";
+        server = HttpServer.create(addr, 0);
         server.createContext("/test", new TestHandler());
-        ExecutorService executor = Executors.newCachedThreadPool();
-        server.setExecutor (executor);
+        executor = Executors.newCachedThreadPool();
+        server.setExecutor(executor);
         server.start();
 
-        URI uri = new URI("http", null,
-                server.getAddress().getHostString(),
+        uri = new URI("http", null,
+                loopback.getHostName(),
                 server.getAddress().getPort(),
                 path, null, null);
+    }
 
-        try {
-            HttpClient client = HttpClient.newHttpClient();
-            HttpRequest request = HttpRequest.newBuilder(uri)
-                    .GET()
-                    .build();
-            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
-            // verify empty response received, otherwise an error has occurred
-            if (!response.body().isEmpty())
-                fail(response.body());
-        } finally {
-            server.stop(2);
-            executor.shutdown();
-        }
+    @Test
+    public void testSend() throws Exception {
+        HttpClient client = HttpClient.newHttpClient();
+        HttpRequest request = HttpRequest.newBuilder(uri)
+                .GET()
+                .build();
+        HttpResponse<String> response = client.send(request, BodyHandlers.ofString());
+        // verify empty response received, otherwise an error has occurred
+        if (!response.body().isEmpty())
+            fail(response.body());
+    }
+
+    @AfterTest
+    public void tearDown() {
+        server.stop(2);
```

Comment:
we usually call `server.stop(0)` in tests.

Reply:
Parameter changed in commit https://github.com/openjdk/jdk/pull/1014/commits/927b2ca213473825b7492c77db23055c55a9768d